src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -783,6 +783,95 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     })
   }
 
+  override def getResourceParent(resource: FullyQualifiedResourceId, samRequestContext: SamRequestContext): IO[Option[FullyQualifiedResourceId]] = {
+    val r = ResourceTable.syntax("r")
+    val rt = ResourceTypeTable.syntax("rt")
+    val pr = ResourceTable.syntax("pr")
+    val prt = ResourceTypeTable.syntax("prt")
+
+    runInTransaction("getResourceParent", samRequestContext)({ implicit session =>
+      val query = samsql"""select ${pr.result.name}, ${prt.result.name}
+              from ${ResourceTable as r}
+              join ${ResourceTypeTable as rt} on ${r.resourceTypeId} = ${rt.id}
+              join ${ResourceTable as pr} on ${pr.id} = ${r.resourceParentId}
+              join ${ResourceTypeTable as prt} on ${pr.resourceTypeId} = ${prt.id}
+              where ${r.name} = ${resource.resourceId}
+              and ${rt.name} = ${resource.resourceTypeName}"""
+
+      import SamTypeBinders._
+      query.map(rs =>
+          FullyQualifiedResourceId(
+            rs.get[ResourceTypeName](prt.resultName.name),
+            rs.get[ResourceId](pr.resultName.name)))
+        .single.apply()
+    })
+  }
+
+  override def setResourceParent(childResource: FullyQualifiedResourceId, parentResource: FullyQualifiedResourceId, samRequestContext: SamRequestContext): IO[Unit] = {
```

Comment:
reworking this query a bit as I realized it allows for a resource to be the parent of itself oops